fsspec/implementations/local.py

```diff
@@ -63,34 +63,34 @@ def glob(self, path, **kwargs):
         return super().glob(path, **kwargs)
 
     def info(self, path, **kwargs):
+        t = None
         if isinstance(path, os.DirEntry):
             # scandir DirEntry
             out = path.stat(follow_symlinks=False)
             link = path.is_symlink()
+            if link:
+                t = "link"
             if path.is_dir(follow_symlinks=False):
                 t = "directory"
             elif path.is_file(follow_symlinks=False):
                 t = "file"
-            else:
-                t = "other"
             path = self._strip_protocol(path.path)
         else:
             # str or path-like
             path = self._strip_protocol(path)
             out = os.stat(path, follow_symlinks=False)
             link = stat.S_ISLNK(out.st_mode)
             if link:
-                out = os.stat(path, follow_symlinks=True)
+                out = os.lstat(path)
+                t = "link"
             if stat.S_ISDIR(out.st_mode):
                 t = "directory"
             elif stat.S_ISREG(out.st_mode):
                 t = "file"
-            else:
-                t = "other"
```

Comment:
The path might be a special dev file (e.g. see S_ISBLK/S_ISFIFO/etc) still, so we shouldn't remove this.

Reply:
It would be handled in return statement